### ISSUE TITLE
score calculator in DeepMCDD on line 66 of train_deepmcdd_table.py "scores = - dists + model.alphas"

### ISSUE THREAD

**Issue by prinxc25** [donalee/DeepMCDD#3]
Good afternoon, Sir. I was going through the code of "MLP_DeepMCDD" algorithm. When using MLP net for tabular data I found that there is use of alphas in calculating score for each input data viz. scores = - dists + model.alphas in line 86 of train_deepmcdd_table.py. Where as per line 102 in DeepMCDD/models/mlp.py
dists = torch.sum((out - self.centers)**2, dim=2) / 2 / torch.exp(2 * F.relu(self.logsigmas)) + self.latent_size * F.relu(self.logsigmas)
1. I couldn't figure out what alpha indicates here.
2.  I found in your paper in section 3.1, paragraph below equation 4 that exp(logsigmas) used in code is radii equivalent for cluster. So, why is it that scores  is  "scores =  - dists + model.alphas" and not "scores =  - dists  + exp(model.logsigmas)".
It will be really helpful Sir, if you can help me understand the calculation of scores.
Waiting eagerly for you reply, thank you.

**Comment by donalee** [comment]
Hi, Thanks for your attention :-)

1. The scores used in "models/mlp.py" is the distance D_k(x) (i.e., class-conditional likelihood) defined in Equation (4).
2. The scores used in "train_deepmcdd_table.py" is the **classification score** P(y=k|x) (=-D_k(x) + b_k) in Equation (6). That is, the alpha indicates the class bias b_k. Note that this score is used for optimizing the network (i.e., computing the second term in Equation (5)). 
3. As you mentioned, exp(model.logsigma) is standard deviation (or radii). It was already used to calculate the dists.

Hope you find this helpful.

**Comment by prinxc25** [comment]
Thank you, so much, Sir.
